core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java

```diff
@@ -509,84 +529,89 @@ private void serializeSeries(
     boolean was_event = false;
     boolean was_event_group = false;
     for (final TypedTimeSeriesIterator<? extends TimeSeriesDataType> iterator : series.iterators()) {
-      while (iterator.hasNext()) {
-        TimeSeriesValue<? extends TimeSeriesDataType> value = iterator.next();
-        if (iterator.getType() == StatusType.TYPE) {
-          if (!was_status) {
-            was_status = true;
-          }
-          json.writeStartObject();
-          writeStatus((StatusValue) value, json);
-          wrote_values = true;
-        } else if (iterator.getType() == EventType.TYPE) {
-          was_event = true;
-          json.writeStartObject();
-          json.writeObjectFieldStart("EventsType");
-          writeEvents((EventsValue) value, json);
-          json.writeEndObject();
-          wrote_values = true;
-        } else if (iterator.getType() == EventGroupType.TYPE) {
-          was_event_group = true;
-          json.writeStartObject();
-          writeEventGroup((EventsGroupValue) value, json, id);
-          wrote_values = true;
-        } else {
-          if (value == null) {
-            continue;
-          }
-          
-          if (iterator.getType() == NumericType.TYPE) {
-            if (writeNumeric((TimeSeriesValue<NumericType>) value, options, 
-                  iterator, json, result, start, end, wrote_values)) {
-              wrote_values = true;
+
+      lock.writeLock().lock(); // since json is not thread safe and we need to form the json in order
```

Comment:
lets do a null check and have the lock null if we're not parallelling.

Reply:
incase of sequential, one thread shouldn't have an impact on the locking.